test/test_abi

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+# This test helps to keep libnxz's ABI stability. It does the following checks:
+#   1. Make sure that libnxz provides all zlib symbols
+#   2. Guarantee that new symbols are not added/removed by mistake
+
+# Exit on first failure
+set -e
+
+# List of zlib exported symbols currently not implemented by libnxz. These
+# must be ignored in our ABI comparison for now.
+missing_symbols=(
+    crc32_z
+    deflateGetDictionary
+    deflateParams
+    deflatePending
+    deflatePrime
+    deflateTune
+    get_crc_table
+    gzbuffer
+    gzclearerr
+    gzclose
+    gzclose_r
+    gzclose_w
+    gzdirect
+    gzdopen
+    gzeof
+    gzerror
+    gzflush
+    gzfread
+    gzfwrite
+    gzgetc
+    gzgetc_
+    gzgets
+    gzoffset64
+    gzoffset
+    gzopen
+    gzopen64
+    gzprintf
+    gzputc
+    gzputs
+    gzread
+    gzrewind
+    gzseek
+    gzseek64
+    gzsetparams
+    gztell
+    gztell64
+    gzungetc
+    gzvprintf
+    gzwrite
+    inflateBack
+    inflateBackEnd
+    inflateBackInit_
+    inflateCodesUsed
+    inflateGetDictionary
+    inflateMark
+    inflatePrime
+    inflateSync
+    inflateUndermine
+    inflateValidate
+    zError
+    zlibCompileFlags
+)
+
+# These should be either the .so files or their descriptions generated by abidw
+zlib_abi_orig="libz.1.2.11.abi"
+libnxz_abi="libnxz.0.63.0.abi"
+libnxz="$1"
+
+
+# Currently abidiff does not allow to ignore SONAME changes, which means it will
+# always spot changes when comparing libnxz and libz. To avoid this, let's
+# artificially change libz's SONAME to suppress this error
+zlib_abi="${zlib_abi_orig}.mod"
+sed "s/soname='libz.so.1'/soname='libnxz.so.0'/g" ${zlib_abi_orig} > ${zlib_abi}
+
+supps_file="zlib.supps"
+trap "rm -f ${supps_file} ${zlib_abi}" EXIT
+
+# Generate suppressions file
+echo > ${supps_file}
```

Comment:
Any reason for generating this dynamically, instead of maintaining it in the repository?

Reply:
It was an attempt to make it more self-contained. I'm OK with putting it in a separate file as well